app/src/androidTest/java/com/deque/mobile/axedevtoolssampleapp/InstrumentationRegistryExampleTest.kt

```diff
@@ -15,52 +20,55 @@ class InstrumentationRegistryExampleTest {
     @JvmField
     val rule: ActivityScenarioRule<MainActivity> = ActivityScenarioRule(MainActivity::class.java)
 
-
     private val axe = AxeDevTools()
 
     init {
         axe.loginWithApiKey(BuildConfig.AXE_DEVTOOLS_APIKEY)
-    }
+         }
 
     @Before
     fun setupAxeDevTools() {
+        axe.tagScanAs(setOf("myAxeEspressoTest"))
+        axe.setTestingConfig(
+            AxeDevToolsEspressoConfig(IdlingRegistry.getInstance())
+        )
         axe.setInstrumentation(InstrumentationRegistry.getInstrumentation())
     }
 
     @Test
-    fun test_using_registry() {
-        axe.setInstrumentation(InstrumentationRegistry.getInstrumentation())
-        val scanResultHandler = axe.scan()
-//      1. Upload it to the dashboard
+    fun test_using_activity() {
+        rule.scenario.onActivity {
+            val scanResultHandler = axe.scan(it)
+            processAccessibilityResults(scanResultHandler)
+        }
+    }
+
+    private fun processAccessibilityResults(scanResultHandler: ScanResultHandler?) {
+        var passes = 0
+        var fails = 0
+        var incomplete = 0
+        val result: com.deque.axe.android.AxeResult? = scanResultHandler?.serializedResult
         scanResultHandler?.uploadToDashboard()
+        scanResultHandler?.saveResultToLocalStorage("axe")
 
-//      2. Using the results in your test suite
-//        var passes = 0
-//        var fails = 0
-//        var incomplete = 0
-//        val result: AxeResult? = scanResultHandler?.serializedResult
-//        result?.axeRuleResults?.forEach { ruleResult ->
-//            when (ruleResult.status) {
-//                AxeStatus.PASS -> {
-//                    passes++
-//                }
-//
-//                AxeStatus.FAIL -> {
-//                    fails++
-//                }
-//
-//                AxeStatus.INCOMPLETE -> {
-//                    incomplete++
-//                }
-//            }
-//        }
-//        assertEquals(6, fails)
+        result?.axeRuleResults?.forEach { ruleResult ->
+            when (ruleResult.status) {
+                AxeStatus.PASS -> {
+                    passes++
+                }
 
-//        3. Save the result JSON to a local file for later use
-        scanResultHandler?.saveResultToLocalStorage("axe")
```

Comment:
Recommendation: The purpose of these comments is to guide the user to choose from different options. So I would recommend keeping this comment.